search-app/src/cmr/search/api/tags.clj

```diff
@@ -23,16 +24,38 @@
   [data]
   (some #(contains? % :errors) data))
 
+(defn tag-association-results->status-code
+  "Check for concept-types requiring error status to be returned.
+  If the concept-type is error-sensitive the function will check for any errors in the results.
+  Will return:
+  - 200 OK -- if response has no errors
+  - 207 MULTI-STATUS -- if response has some errors and some successes
+  - 400 BAD REQUEST -- if response has all errors"
+  [results]
+    (let [result-count (count results)
+          num-errors (assoc/num-errors-in-assoc-results results)]
+      (if (= 0 result-count)
+        200
+        (if (= num-errors result-count)
+          400
+          (if (> num-errors 0)
+            207
+            200)))))
+
```

Comment:
a lot of nested if statements. I don't think you need the first one, your last one assumes 200, so you only really testing for 400 and 207. Also, are these constants somewhere in ring? I wonder if the one (if it exists) for 207 is descriptive.